### ISSUE TITLE
`ReferenceError: ReactIcon is not defined` on Vanilla Mastodon

### ISSUE THREAD

**Issue by Thanh9356** [neatchee/mastodon#82]
### Steps to reproduce the problem

1. Set Mastodon *Flavors Settings* to *Vanilla Mastodon*
2. Return back to the main website
3. Error! I can't move to anything else :\

### Expected behaviour

The Home page should be used normally, anything else too.

### Actual behaviour

The site says it has error, and moving doesn't help. Switching to Glitch Mastodon then it works again

### Mastodon instance

urusai.social

### Mastodon version

v4.3.0-alpha.1+glitch

### Browser name and version

Firefox 122.0.1

### Operating system

openSUSE Tumbleweed, Android 8.1.0

### Technical details

Error code copied from website:
```
ReferenceError: ReactIcon is not defined
iconComponent (webpack:///app/javascript/home/mastodon/live/app/javascript/mastodon/components/status_action_bar.jsx:405:8)
kj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:204:188)
ij (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:203:148)
null,Zf(d)?(f=!0,cg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:292:171)
uj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:280:391)
uj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:280:322)
uj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:280:179)
Ok (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:271:87)
Hk (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:268:428)
J (webpack:///node_modules/scheduler/cjs/scheduler.production.min.js:13:204)
S (webpack:///node_modules/scheduler/cjs/scheduler.production.min.js:14:127)
run (webpack:///node_modules/setimmediate/setImmediate.js:40:12)
setTimeout (webpack:///node_modules/setimmediate/setImmediate.js:69:24)
event.data.indexOf (webpack:///node_modules/setimmediate/setImmediate.js:109:16)
```
Screenshot:
![image](https://github.com/neatchee/mastodon/assets/36446935/d61632da-3f65-4587-945d-0fedabc51fa9)

**Comment by neatchee** [comment]
Well this kinda stinks. They decided to roll back the emoji integration from the vanilla flavour, so it's a glitch-only feature now. I'm going to have to unwind this.... :\

**Comment by neatchee** [comment]
Should be resolved by 231c6ce19376124b28d3e1f89c54bc02e47a6d71
Please check for any other issues in the Vanilla flavour; I do not use it and I don't think many of my users do either
